### ISSUE TITLE
Release UNION with code and pre-trained checkpoints on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [TedLentsch/UNION#2]
Hi @TedLentsch 🤗

I'm Niels and work as part of the open-source team at Hugging Face. I discovered your work on Arxiv and noticed that it was just added to the Hugging Face paper page at https://huggingface.co/papers/2405.15688.

I wanted to reach out to link the 3D object detection checkpoints that you have released to the paper page to improve their discoverability. This will let people find the models easier when filtering https://huggingface.co/models, and it will also help the models get more visibility.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by TedLentsch** [comment]
Hi @NielsRogge,

Thanks for getting in touch.

Our UNION method auto-labels nuScenes raw sensor data to produce 3D pseudo-bounding box annotations, which we then use to train existing 3D detectors. As Table 2 in our [paper](https://arxiv.org/pdf/2405.15688) shows, these pseudo-labels fall short of the official ground truth in detection performance but provide a scalable way to expand training data.

Since our contribution is in labeling rather than new model architectures, I agree it makes sense to publish our pseudo-labels for nuScenes. We will host them in a new Hugging Face Dataset repository (licensed CC BY-NC-SA 4.0 to match the original nuScenes terms). I will look into this soon and close this issue for now.

Best,

Ted Lentsch

**Comment by NielsRogge** [comment]
Ok great, thanks for your extensive reply.

Feel free to keep this issue open so we can track it :)
